src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -337,4 +338,57 @@ private static async Task PeriodAsync(int period, int iterations)
             await tcs.Task.ConfigureAwait(false);
         }
     }
+
+    [Fact]
+    public void TimersCreatedConcurrentlyOnDifferentThreadsAllFire()
+    {
+        int processorCount = Environment.ProcessorCount;
+
+        int timerTickCount = 0;
+        TimerCallback timerCallback = data => Interlocked.Increment(ref timerTickCount);
+
+        var threadStarted = new AutoResetEvent(false);
+        var createTimers = new ManualResetEvent(false);
+        var timers = new Timer[processorCount];
+        Action<object> createTimerThreadStart = data =>
+        {
+            int i = (int)data;
+            var sw = new Stopwatch();
+            threadStarted.Set();
+            createTimers.WaitOne();
+
+            // Use the CPU a bit around creating the timer to try to have some of these threads run concurrently
+            sw.Restart();
```

Comment:
nit, you can do one line `var sw= Stopwatch.StartNew()`

Reply:
I wanted to get the allocation out of the way before the test starts to have more predictable timing during the test